k8s/autopilot/autopilot.go

```diff
@@ -0,0 +1,123 @@
+package autopilot
+
+import (
+	"fmt"
+	"os"
+
+	autopilotclientset "github.com/libopenstorage/autopilot-api/pkg/client/clientset/versioned"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var (
+	instance Client
+)
+
+// Ops provides an interface to Autopilot operations.
+type Ops interface {
+	AutopilotRuleOps
+
+	// SetConfig sets the config and resets the client
+	SetConfig(config *rest.Config)
+}
+
+// Instance returns a singleton instance of the autopilot client.
+func Instance() Ops {
+	return &instance
+}
+
+func New(c autopilotclientset.Interface) *Client {
+	return &Client{
+		autopilot: c,
+	}
+}
+
+// NewForConfig creates a new autopilot client for the given config.
+func NewForConfig(cfg *rest.Config) (*Client, error) {
```

Comment:
Actually do we need the New and NewForConfig method? It's basically an addition to the Instance() and SetConfig() + Instance() methods, which will return new instances instead of the singleton.

Reply:
Actually we will need a SetClient() for testing where we can pass a fake client to the singleton.

Reply:
New/NewForConfig returns a new instance of the client while Instance() returns the same one each time.

Would it be better to add a `SetInstance(client Ops)` function to the package (no need to add this to interface as it's for testing purpose)?